mesonbuild/compilers/c.py

```diff
@@ -420,6 +420,11 @@ def get_options(self) -> 'OptionDictType':
     def get_option_compile_args(self, options: 'OptionDictType') -> T.List[str]:
         args = []
         std = options['std']
+        if std.value.startswith('gnu'):
+            mlog.warning(
+                'cl.exe does not actually support gnu standards, and meson '
+                'will instead demote to the nearest ISO C standard. This '
+                'may cause compilation to fail.', once=True)
```

Comment:
We should not add this warning till we have a replacement for this workflow.

Please, this is really really bad. We should **never** be spammy about warnings. It makes `--fatal-meson-warnings` unusable by projects and makes it harder for us to deprecate things because people just ignore warnings.

Reply:
I've replaced the warning with a log() message instead, as the message seems useful to me (all of my projects that use gnu99 will fail to compile with c99, so giving me *something* is nice.) I've also replaced a similar warning in the IntelCl compiler with a log() message instead.

I'll get to adding a fallback mechanism later, but for now is this sufficient to land?